File: server/r/api/pri/compose_api/cm.go
/*
 * Copyright (C) 2019 The Qing Project. All rights reserved.
 *
 * Use of this source code is governed by a license that can
 * be found in the LICENSE file.
 */

package composeapi

import (
	"qing/a/def/appDef"
	"qing/lib/clib"

	"github.com/mgenware/goutil/jsonx"
)

type PostCoreContentLoader struct {
	dict map[string]any
}

func NewPostCoreContentLoader(dict map[string]any) PostCoreContentLoader {
	return PostCoreContentLoader{dict: dict}
}

func (v PostCoreContentLoader) MustGetTitle() string {
	return clib.MustGetStringFromDict(v.dict, "title", appDef.LenMaxTitle)
}

func (v PostCoreContentLoader) MustGetSummary() string {
	return clib.MustGetTextFromDict(v.dict, "summary")
}

func (v PostCoreContentLoader) MustGetHTML() string {
	return clib.MustGetTextFromDict(v.dict, "html")
}

func (v PostCoreContentLoader) GetOptionalSrc() *string {
	return jsonx.GetStringOrNil(v.dict, "src")
}

File: it/br/src/profile/cm.ts
/*
 * Copyright (C) 2023 The Qing Project. All rights reserved.
 *
 * Use of this source code is governed by a license that can
 * be found in the LICENSE file.
 */

import { expect } from '@playwright/test';
import * as br from 'br.js';

export interface CheckProfileFeedArgs {
  title: string;
  link: string;
}

export async function checkProfileFeed(el: br.BRElement, e: CheckProfileFeedArgs) {
  await expect(el.$a({ href: e.link, text: e.title }).c).toBeVisible();
}
